tests/unit/test_req_install.py

```diff
@@ -108,3 +109,26 @@ def test_install_req_from_string_with_comes_from_without_link(self):
         assert install_req.link.url == wheel_url
         assert install_req.req.url == wheel_url
         assert install_req.is_wheel
+
+    def test_install_req_vcs_without_wheel_warning(self, caplog):
+        """
+        Test to make sure that installing from VCS without wheel generates
+        a warning.
+        """
+        req = InstallRequirement(
+            Requirement("gputil==1.4.0"),
+            comes_from=[],
+            link=Link("git+https://github.com/llamafilm/gputil")
+        )
+
+        req.source_dir = os.curdir
+        req.install([])
```

Comment:
Uhhh, don't do this in a unit test... This has side effects that affect the interpreter it's running in. Specifically, it'll install the package when you run the tests

Reply:
What is the correct way to do it? This is how it was done in `test_req_file.py`

Reply:
lemme know what you think @pradyunsg

Reply:
ping @pradyunsg

Reply:
Thanks for the ping! This got buried in my notifications. :(

Write an integration test that exercises this code path. Here's an example, that you'll likely need to change the name of the package for. Feel free to add the source code for a minimal reproducer package in `tests/data/packages`.

```py
def test_install_distribution_union_with_constraints(
    script,
    data,
):
    to_install = data.packages.joinpath("a-thing-that-triggers-this")
    result = script.pip_install_local(
        to_install,
        allow_stderr_warning=True,
    )
    msg = "Direct URL of package 'a-thing-that-triggers-this' will not be recorded when using"
    assert msg in result.stderr
```

Reply:
See `test_basic_install_editable_from_svn` for an example of a local-only VCS test.